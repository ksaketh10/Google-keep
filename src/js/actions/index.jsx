import { ADD_TASK } from "../constants/index";

export function addTask(task) {
    return{ type : ADD_TASK, task }
}

export function getData() {
    return { type: "DATA_REQUESTED" };
  }