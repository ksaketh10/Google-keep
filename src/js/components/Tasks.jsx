import React from "react";
import { connect } from "react-redux";
const mapStateToProps = state => {
  console.log(state)
  return { tasks: state.tasks };
};
const tasksList = ({ tasks }) => (
  <ul className="list-group list-group-flush">
    {tasks.map(task => (
      <li className="list-group-item" key={task.title.toString()}>
        <div className="card">
          <div class="card-header">
          <h5 className="card-title">{task.title}</h5>
          </div>
            <div className="card-body">
                <p className="card-text">{task.task}</p>
            </div>
        </div>
      </li>
    ))}
  </ul>
);
const List = connect(mapStateToProps)(tasksList);
export default List;