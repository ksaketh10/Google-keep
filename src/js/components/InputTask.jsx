import { addTask } from "../actions";
import React, { Component } from "react";
import { connect } from "react-redux";

function mapDispatchToProps(dispatch) {
    return {
        addTask: task => dispatch(addTask(task))
    };
}

class InputTask extends Component {
    constructor() {
        super();
        this.state = {
            title: "",
            task: ""
        };
        this.handleSubmit = this.handleSubmit.bind(this)
        this.onChangeTitle = this.onChangeTitle.bind(this)
        this.onChangeTask = this.onChangeTask.bind(this)
    }

    handleSubmit(event) {
        event.preventDefault()
        const { title, task } = this.state
        this.props.addTask({ title, task })
        this.setState({ title: "", task: "" })
    }

    onChangeTitle(event) {
        event.preventDefault()
        this.setState({title : event.target.value, task : this.state.task})
    }

    onChangeTask(event) {
        event.preventDefault()
        this.setState({title : this.state.title, task : event.target.value})
    }

    render() {
        return (
            <form onSubmit={this.handleSubmit}>
                <div className="form-row">
                    <div className="col-12">
                        <input type="text" className="form-control title" placeholder="Title" onChange={this.onChangeTitle}/>
                    </div>
                </div>
                <div className="form-row">
                    <div className="col-12">
                        <textarea type="text" className="form-control task" placeholder="Task" onChange={this.onChangeTask}/>
                    </div>
                </div>
                <div className="form-row mb-4">
                    <div className="col-3">
                        <button type="submit" className="btn btn-primary mb-2 add">Add</button>
                    </div>
                </div>
            </form>
        );
    }
}
const Form = connect(null, mapDispatchToProps)(InputTask);
export default Form;