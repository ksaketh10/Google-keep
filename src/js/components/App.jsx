import React from "react";
import InputTask from "../components/InputTask";
import Tasks from "../components/Tasks";
import Post1  from "./Post";
const App = () => (
    <div className="row mt-5">
        <div className="col-md-4 offset-md-1">
            <h2>Add a new Task</h2>
            <InputTask />
        </div>
        <div className="col-md-4 offset-md-1">
            <h2>Tasks</h2>
            <Tasks />
        </div>
        <div className="col-md-4 offset-md-1">
            <h2>Tasks</h2>
            <Post1 />
        </div>

    </div>
);
export default App;