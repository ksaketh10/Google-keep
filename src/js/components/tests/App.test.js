import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import App from '../App';
import ConnectedInputTask from '../InputTask';
import { InputTask } from '../InputTask';
import ConnectedTasks from '../Tasks'
import { TasksList } from '../Tasks';
import configureStore from 'redux-mock-store'
import { Provider } from "react-redux";
import { addTask } from "../../actions/index"
import renderer from 'react-test-renderer'

configure({ adapter: new Adapter() });

describe('App component', () => {

    const mockStore = configureStore();

    it('2 + 2 is 4', () => {
        expect(2 + 2).toBe(4)
    });

    it('renders page correctly', () => {
        const wrapper = shallow(<App />)
        expect(wrapper.find('InputTask')).toBeDefined();
        expect(wrapper.find('Tasks')).toBeDefined();
        expect(wrapper.find('Post')).toBeDefined();
    })

    it('renders App', () => {
        const wrapper = shallow(
            <App />
        );
        expect(wrapper.contains(<h2>Add a new Task</h2>)).toBe(true)
        expect(wrapper.contains(<ConnectedInputTask />)).toBe(true)
        expect(wrapper.contains(<ConnectedTasks />)).toBe(true)
    })

    it('renders title and task input and button', () => {
        const props = {
            addTask: jest.fn()
        }
        const wrapper = shallow(<InputTask {...props} />)
        expect(wrapper.find('form').length).toEqual(1)
        expect(wrapper.find('input[placeholder="Title"]')).toBeDefined()
        expect(wrapper.find('textarea[placeholder="Task"]')).toBeDefined()
        expect(wrapper.find('button.add').text()).toEqual('Add')
        // expect(wrapper.contains(<input type="text" className="form-control title" placeholder="Title" />)).toBe(true);
        //expect(wrapper.contains(<input type="textarea" className="task form-control" placeholder="Task" />)).toBe(true)
        // expect(wrapper.contains(<button type="submit" className="btn btn-primary mb-2">Add</button>)).toBe(true)
    })

    it('renders dumb component tasks', () => {
        const tasks = [{ 'title': 'title1', 'task': 'task1' }]
        const wrapper = shallow(<TasksList tasks={tasks} />)

        expect(wrapper.contains(<h5 className="card-title">title1</h5>))
        expect(wrapper.contains(<p className="card-text">task1</p>))
        expect(wrapper.find('li').length).toBe(1)
        expect(wrapper.length).toEqual(1)
    })

    it('check action on dispatch', () => {
        // const initialState = {
        //     tasks: [],
        //     remoteArticles: []
        // }
        // let store = mockStore(initialState)

        // const wrapper = shallow(<ConnectedInputTask store={store} />)
        // expect(wrapper.length).toEqual(1)

    })

    it('renders Connected component tasks', () => {
        const initialState = {
            tasks: [],
            remoteArticles: []
        }
        let store = mockStore(initialState)
        const wrapper = shallow(<ConnectedTasks store={store} />)
        expect(wrapper.length).toEqual(1)
        expect(wrapper.find(TasksList).prop('tasks')).toEqual(initialState.tasks)
    })

    it('check action dispatch ', () => {
        const initialState = {
            tasks: [],
            remoteArticles: []
        }
        let store = mockStore(initialState)
        const wrapper = mount(<Provider store={store}>
            <ConnectedTasks store={store} />
        </Provider>)
        store.dispatch(addTask({ 'title': 'title1', 'task': 'task1' }));
        let action = store.getActions()
        expect(action[0].type).toBe("ADD_TASK")
        expect(wrapper.find(TasksList).prop('tasks')).toEqual(initialState.tasks)
    })

    it('click on add check action', () => {
        const mockCallBack = jest.fn();
        const props = {
            addTask: mockCallBack
        }
        const wrapper = shallow(<InputTask  {...props} />)
        const addButton = wrapper.find('button.add')
        addButton.simulate('click')
        expect(mockCallBack.mock.calls.length).toEqual(1)
    })

    it('+++capturing Snapshot of Home', () => {
        const initialState = {
            tasks: [],
            remoteArticles: []
        }
        let store = mockStore(initialState)
        const renderedValue = renderer.create(<Provider store={store}><App /></Provider>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

    // it('Add a task and renders the content', () => {
    //     const wrapper = mount(<Provider store={store}>
    //         <App />
    //     </Provider>)

    //     const title = wrapper.find('input.title')
    //     title.simulate('change', { target: { value: 'Title' } })

    //     const task = wrapper.find('textarea.task')
    //     task.simulate('change', { target: { value: 'Task' } })

    //     // const title = wrapper.find('input').get(0)
    //     // title.value = 'Title'

    //     // const task = wrapper.find('input').get(1)
    //     // task.value = 'Task'

    //     const addButton = wrapper.find('button.add')
    //     addButton.simulate('click')

    //     expect(wrapper.contains(<li class="list-group-item">
    //         <div class="card"><div class="card-header">
    //             <h5 class="card-title">Title</h5>
    //         </div>
    //             <div class="card-body">
    //                 <p class="card-text">Title</p>
    //             </div></div>
    //     </li>)).toBe(true)
    // })
});