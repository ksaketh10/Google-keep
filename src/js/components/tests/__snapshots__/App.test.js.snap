// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component +++capturing Snapshot of Home 1`] = `
<div
  className="row mt-5"
>
  <div
    className="col-md-4 offset-md-1"
  >
    <h2>
      Add a new Task
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-row"
      >
        <div
          className="col-12"
        >
          <input
            className="form-control title"
            onChange={[Function]}
            placeholder="Title"
            type="text"
          />
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className="col-12"
        >
          <textarea
            className="form-control task"
            onChange={[Function]}
            placeholder="Task"
            type="text"
          />
        </div>
      </div>
      <div
        className="form-row mb-4"
      >
        <div
          className="col-3"
        >
          <button
            className="btn btn-primary mb-2 add"
            type="submit"
          >
            Add
          </button>
        </div>
      </div>
    </form>
  </div>
  <div
    className="col-md-4 offset-md-1"
  >
    <h2>
      Tasks
    </h2>
    <ul
      className="list-group list-group-flush"
    />
  </div>
  <div
    className="col-md-4 offset-md-1"
  >
    <h2>
      Tasks
    </h2>
    <ul
      className="list-group list-group-flush"
    />
  </div>
</div>
`;
